src/Host.cpp

```diff
@@ -369,6 +369,28 @@ void Host::resetProfile()
     qDebug()<<"resetProfile() DONE";
 }
 
+// Saves profile to disk - does not save items dirty in the editor, however.
+// returns true if successful or false+error message if not
+std::tuple<bool, QString, QString, QString, QString> Host::saveProfile()
```

Comment:
Is this only to be used from the Lua sub-system or is it intended to use it from elsewhere in the C++ core? The `Host::name` value returned from `Host::getName()` is always going to be available won't it?  (We have `getProfileName()` now in Lua!)

Given that if it is reused elsewhere we may not be interested in the file name or error message, I think this is one of those cases where pointers to `QStrings` which have a nullptr default value is a valid approach to use, so that this might become:

    bool Host::saveProfile( QString * pPathFileName = nullptr, QString * pErrMsg = nullptr );

I would aim for something that:
+ If `pPathFileName` is a `nullptr` or is a pointer to an _empty_ `QString` then generate a date/time stamp form pathFileName and for the second case, store that in the pointed to location.
+ if `pPathFileName` does point to a non-empty QString then, if it is relative, resolve it to either the profile home directory (or the "/current" ?) subdirectory or if it is absolute then treat it as such.
+ if the resolved saved directory does not exist then try to create it (and all its ancestor directories) - if this is not possible then return a `false` value and, if `pErrMsg` is not null stick the system error message in there.
+ try to open and write the Host XML - return `false` and the system provided open or write file message if either step fails in `pErrMsg` if it exists.
+ do the `saveModules(1)`
+ return `true` as we have succeeded and populated the save file name if appropriate, and left the error message untouched because there is no error.

I would comment that I'm _sure_ I experimented with code like this in the past - and I used the pErrMsg or equivalent as being nullptr as the flag to generate "[ XXX ] - " messages in the main console - I'd only populate a valid pErrMsg with a locally generated message if there is NOT a system one to hand!